spec/models/service_template/copy_spec.rb

```diff
@@ -0,0 +1,265 @@
+        @st1.add_resource(FactoryBot.create(:configuration_script_base))
+        expect(ServiceTemplate.count).to eq(1)
+        @st1.template_copy("new_template")
+        new_service_template = ServiceTemplate.find_by(:name => "new_template")
+        expect(ServiceTemplate.count).to eq(2)
+        expect(@st1.service_resources.first.resource).not_to be(nil)
+        expect(new_service_template.service_resources.first.resource).to eq(@st1.service_resources.first.resource)
+        expect(ConfigurationScriptBase.count).to eq(1)
+        expect(new_service_template.display).to be(false)
+        expect(new_service_template.guid).not_to eq(@st1.guid)
+      end
+
+      it "with non-copyable resource (ext management system)" do
+        @st1.add_resource(FactoryBot.create(:ext_management_system))
+        expect(ServiceTemplate.count).to eq(1)
+        @st1.template_copy("new_template")
+        new_service_template = ServiceTemplate.find_by(:name => "new_template")
+        expect(ServiceTemplate.count).to eq(2)
+        expect(new_service_template.service_resources.first.resource_id).to eq(@st1.service_resources.first.resource_id)
+        expect(ExtManagementSystem.count).to eq(1)
+        expect(new_service_template.guid).not_to eq(@st1.guid)
+        expect(new_service_template.display).to be(false)
+        expect(@st1.service_resources.first.resource).not_to be(nil)
+      end
+
+      it "with non-copyable resource (orchestration template)" do
+        @st1.add_resource(FactoryBot.create(:orchestration_template))
+        expect(ServiceTemplate.count).to eq(1)
+        @st1.template_copy("new_template")
+        new_service_template = ServiceTemplate.find_by(:name => "new_template")
+        expect(ServiceTemplate.count).to eq(2)
+        expect(new_service_template.service_resources.first.resource_id).to eq(@st1.service_resources.first.resource_id)
+        expect(OrchestrationTemplate.count).to eq(1)
+        expect(new_service_template.guid).not_to eq(@st1.guid)
+        expect(new_service_template.display).to be(false)
+        expect(@st1.service_resources.first.resource).not_to be(nil)
+      end
+
+      it "with copyable resource" do
+        admin = FactoryBot.create(:user_admin)
+        vm_template = FactoryBot.create(:vm_openstack, :ext_management_system => FactoryBot.create(:ext_management_system))
+        ptr = FactoryBot.create(:miq_provision_request_template, :requester => admin, :src_vm_id => vm_template.id)
+        @st1.add_resource(ptr)
+        expect(ServiceTemplate.count).to eq(1)
+        @st1.template_copy("new_template")
+        new_service_template = ServiceTemplate.find_by(:name => "new_template")
+        expect(ServiceTemplate.count).to eq(2)
+        expect(MiqProvisionRequestTemplate.count).to eq(2)
+        expect(new_service_template.guid).not_to eq(@st1.guid)
+        expect(new_service_template.display).to be(false)
+        expect(new_service_template.service_resources).not_to be(nil)
```

Comment:
Since this is a relation it will never be `nil`, right?  It should always be a `ActiveRecord::Associations::CollectionProxy`

Reply:
Nevermind, you didn't change this...